samza-core/src/main/java/org/apache/samza/execution/StreamEdge.java

```diff
@@ -54,8 +54,12 @@ void addTargetNode(ProcessorNode targetNode) {
     targetNodes.add(targetNode);
   }
 
-  StreamSpec getStreamSpec() {
-    return streamSpec;
+  public StreamSpec getStreamSpec() {
+    if (partitions == PARTITIONS_UNKNOWN) {
+      return streamSpec;
+    } else {
+      return streamSpec.copyWithPartitionCount(partitions);
```

Comment:
There is only one caller to this method and it's on the code path I modified/verified. :-)